lntest/harness.go

```diff
@@ -479,37 +479,137 @@ func (h *HarnessTest) SetTestName(name string) {
+	// this field must be encoded as base64.
+	walletPassword []byte
+}
+
+// NewNodeOpt is a functional option for NewNode.
+type NewNodeOpt func(*newNodeOpts)
+
+// WithCoins is a functional option that funds the node with coins after
+// creation.
+func WithCoins() NewNodeOpt {
+	return func(opts *newNodeOpts) {
+		opts.withCoins = true
+	}
+}
+
+// WithSeedEntropy returns a functional option that injects explicit entropy
+// into the node, enabling deterministic control over mnemonic seed derivation.
+func WithSeedEntropy(seedEntropy []byte, walletPassword []byte) NewNodeOpt {
+	return func(opts *newNodeOpts) {
+		opts.seedEntropy = seedEntropy
+		opts.walletPassword = walletPassword
+	}
+}
+
 // NewNode creates a new node and asserts its creation. The node is guaranteed
 // to have finished its initialization and all its subservers are started.
-func (h *HarnessTest) NewNode(name string,
+func (h *HarnessTest) NewNode(name string, extraArgs []string,
+	nodeOpts ...NewNodeOpt) *node.HarnessNode {
+
+	// Apply functional options.
+	opts := &newNodeOpts{}
+	for _, opt := range nodeOpts {
+		opt(opts)
+	}
+
+	var hNode *node.HarnessNode
+
+	// If a seed is provided, use the seed-based creation flow.
+	if len(opts.seedEntropy) > 0 {
+		hNode = h.createNodeWithSeed(name, extraArgs, opts)
+	} else {
+		// Otherwise, use the regular node creation flow.
+		hNode = h.createRegularNode(name, extraArgs)
+	}
+
+	// If withCoins option is set, fund the node with coins.
+	if opts.withCoins {
+		h.fundNodeWithCoins(hNode)
+	}
+
+	return hNode
+}
+
+// createRegularNode creates a node using the regular flow (no seed).
+func (h *HarnessTest) createRegularNode(name string,
 	extraArgs []string) *node.HarnessNode {
 
-	node, err := h.manager.newNode(h.T, name, extraArgs, nil, false)
+	hNode, err := h.manager.newNode(h.T, name, extraArgs, nil, false)
 	require.NoErrorf(h, err, "unable to create new node for %s", name)
 
 	// Start the node.
-	err = node.Start(h.runCtx)
-	require.NoError(h, err, "failed to start node %s", node.Name())
+	err = hNode.Start(h.runCtx)
+	require.NoError(h, err, "failed to start node %s", hNode.Name())
 
 	// Get the miner's best block hash.
 	bestBlock, err := h.miner.Client.GetBestBlockHash()
 	require.NoError(h, err, "unable to get best block hash")
 
 	// Wait until the node's chain backend is synced to the miner's best
 	// block.
-	h.WaitForBlockchainSyncTo(node, *bestBlock)
+	h.WaitForBlockchainSyncTo(hNode, *bestBlock)
 
-	return node
+	return hNode
 }
 
-// NewNodeWithCoins creates a new node and asserts its creation. The node is
-// guaranteed to have finished its initialization and all its subservers are
-// started. In addition, 5 UTXO of 1 BTC each are sent to the node.
-func (h *HarnessTest) NewNodeWithCoins(name string,
-	extraArgs []string) *node.HarnessNode {
+// createNodeWithSeed creates a node using the seed-based flow.
+func (h *HarnessTest) createNodeWithSeed(name string, extraArgs []string,
```

Comment:
I think we already have `newNodeWithSeed`?

Reply:
So this is similar to the method, which I think we should combine them,
https://github.com/lightningnetwork/lnd/blob/a5c4a7c547023ad5c74ab01f71ddff53e9d0fe1f/lntest/harness.go#L628-L633

Reply:
let's fix this and merge?